builtin/providers/aws/resource_aws_dc_intra_virtual_interface_confirm.go

```diff
@@ -0,0 +1,216 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDirectConnectIntraVirtualInterfaceConfirm() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDirectConnectIntraVirtualInterfaceConfirmCreate,
+		Read:   resourceAwsDirectConnectIntraVirtualInterfaceConfirmRead,
+		Delete: resourceAwsDirectConnectIntraVirtualInterfaceConfirmDelete,
+
+		Schema: map[string]*schema.Schema{
+
+			"interface_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"virtual_interface_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"virtual_gateway_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"allow_down_state": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceAwsDirectConnectIntraVirtualInterfaceConfirmCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dcconn
+
+	var err error
+
+	if v, ok := d.GetOk("interface_type"); ok && v.(string) == "public" {
+
+		createOpts := &directconnect.ConfirmPublicVirtualInterfaceInput{
+			VirtualInterfaceId: aws.String(d.Get("virtual_interface_id").(string)),
+		}
+
+		log.Printf("[DEBUG] Creating DirectConnect public virtual interface")
+		_, err = conn.ConfirmPublicVirtualInterface(createOpts)
+		if err != nil {
+			return fmt.Errorf("Error creating DirectConnect Virtual Interface: %s", err)
+		}
+
+	} else {
+
+		createOpts := &directconnect.ConfirmPrivateVirtualInterfaceInput{
+			VirtualInterfaceId: aws.String(d.Get("virtual_interface_id").(string)),
+		}
+
+		if v, ok := d.GetOk("virtual_gateway_id"); ok {
+			createOpts.VirtualGatewayId = aws.String(v.(string))
+		} else {
+			return fmt.Errorf("Error virtual_gateway_id is required for private virtual interface with id: %s", d.Get("virtual_interface_id").(string))
+		}
+
+		log.Printf("[DEBUG] Creating DirectConnect private virtual interface")
+		_, err = conn.ConfirmPrivateVirtualInterface(createOpts)
+		if err != nil {
+			return fmt.Errorf("Error creating DirectConnect Virtual Interface: %s", err)
+		}
+
+	}
+
+	var stateConf *resource.StateChangeConf
+
+	if v, ok := d.GetOk("allow_down_state"); ok && v.(bool) {
+
+		stateConf = &resource.StateChangeConf{
```

Comment:
I think we can handle this slightly differently. Set a []string depending on the `allow_down_state` variable

Then use the stateConf once and pass in the correctly populated variable